pastas/stats/signatures.py

```diff
@@ -0,0 +1,1353 @@
+    and the median head.
+
+    References
+    ----------
+    .. [baker_2004] Baker, D. B., Richards, P., Loftus, T. T., & Kramer,
+       J. W. (2004). A new flashiness index: Characteristics and applications
+       to midwestern rivers and streams. Journal of the American Water
+       Resources Association, 40(2), 503–522.
+
+    """
+    if normalize:
+        series = _normalize(series)
+
+    series = series.dropna()
+    dt = diff(series.index.to_numpy()) / Timedelta("1D")
+    dh = series.diff().dropna()
+    # sum(dt) is more fair with irregular time series
+    return sum(sqrt(dh ** 2 + dt ** 2)) / sum(dt)
+
+
+def richards_baker_index(series, normalize=True):
+    """Richards-Baker index according to [baker_2004]_.
+
+    Parameters
+    ----------
+    series: pandas.Series
+        Pandas Series with DatetimeIndex and head values.
+    normalize: bool, optional
+        normalize the time series to values between zero and one.
+
+    Returns
+    -------
+    float
+
+    Notes
+    -----
+    Sum of absolute values of day‐to‐day changes in head divided by the sum
+    of scaled daily head. Equivalent the Richards Pathlength without the
+    time component.
+
+    References
+    ----------
+    .. [baker_2004] Baker, D. B., Richards, P., Loftus, T. T., & Kramer,
+       J. W. (2004). A new flashiness index: Characteristics and applications
+       to midwestern rivers and streams. Journal of the American Water
+       Resources Association, 40(2), 503–522.
+
+    """
+    if normalize:
+        series = _normalize(series)
+
+    return series.diff().dropna().abs().sum() / series.sum()
+
+
+def baseflow_index(series, normalize=True):
+    """Baseflow index according to [wmo_2008]_.
+
+    Parameters
+    ----------
+    series: pandas.Series
+        Pandas Series with DatetimeIndex and head values.
+    normalize: bool, optional
+        normalize the time series to values between zero and one.
+
+    Returns
+    -------
+    float
+
+    Notes
+    -----
+    Adapted analogously to its application in streamflow. Here, a baseflow
+    time series is separated from a 5‐day minimum groundwater head in a
+    moving window. BFI equals the total sum of heads of original time series
+    divided by the total sum of heads from the baseflow type of time series.
+
+    References
+    ----------
+    .. [wmo_2008] WMO (2008). Manual on Low‐Flow Estimation and Prediction.
+       Geneva, Switzerland: World Meteorological Organization.
+
+    """
+    if normalize:
+        series = _normalize(series)
+
+    # A/B. Selecting minima hm over 5 day periods
+    hm = series.resample("5D").min()
+
+    # C. define the turning points ht (0.9 * head < adjacent heads)
+    ht = pd.Series(dtype=float)
+    for i, h in enumerate(hm.iloc[1:-1], start=1):
+        if (h < hm.iloc[i - 1]) & (h < hm.iloc[i + 1]):
+            ht[hm.index[i]] = h
+
+    # D. Interpolate
+    ht = ht.resample("D").interpolate()
```

Comment:
[TypeError: Only valid with DatetimeIndex, TimedeltaIndex or PeriodIndex, but got an instance of 'Index'](https://github.com/pastas/pastas/pull/372/files)

Seems like `ht` is missing a datetime index so the resampling does not work.